core/kazoo_bindings/src/kazoo_bindings.erl

```diff
@@ -710,7 +713,9 @@ map_processor_fold(#kz_binding{binding_parts=BParts
                   ,_Routing
                   ,RoutingParts
                   ) ->
-    case matches(BParts, RoutingParts) of
+    case matches(BParts, RoutingParts)
+        orelse matches(RoutingParts, BParts)
```

Comment:
You're right I may have misunderstood how bindings work, and I didn't think of looking at the tests.

Here is the code that made me think I could write tasks_bindings:apply/3:

```
applications/crossbar/src/crossbar_maintenance.erl:432:    Context1 = crossbar_bindings:fold(<<"v1_resource.validate.accounts">>, Payload),
applications/crossbar/src/crossbar_maintenance.erl:459:    Context1 = crossbar_bindings:fold(<<"v1_resource.validate.users">>, Payload),
applications/crossbar/src/crossbar_maintenance.erl:478:    Context1 = crossbar_bindings:fold(<<"v1_resource.execute.put.accounts">>, [Context]),
applications/crossbar/src/crossbar_maintenance.erl:503:    Context1 = crossbar_bindings:fold(<<"v1_resource.execute.put.users">>, [Context]),
applications/crossbar/src/modules/cb_bulk.erl:305:    InterimContext = crossbar_bindings:fold(Binding, Payload),
applications/crossbar/src/modules/cb_onboard.erl:160:    Context1 = crossbar_bindings:fold(<<"*.validate.accounts">>, Payload),
applications/crossbar/src/modules/cb_onboard.erl:191:    Context1 = crossbar_bindings:fold(<<"*.validate.phone_numbers">>, Payload),
applications/crossbar/src/modules/cb_onboard.erl:228:            Context1 = crossbar_bindings:fold(<<"*.validate.braintree">>, Payload),
applications/crossbar/src/modules/cb_onboard.erl:281:    Context1 = crossbar_bindings:fold(<<"*.validate.users">>, Payload),
applications/crossbar/src/modules/cb_onboard.erl:340:    Context1 = crossbar_bindings:fold(<<"*.validate.devices">>, Payload),
applications/crossbar/src/modules/cb_onboard.erl:390:    Context1 = crossbar_bindings:fold(<<"*.validate.vmboxes">>, Payload),
applications/crossbar/src/modules/cb_onboard.erl:434:    Context1 = crossbar_bindings:fold(<<"*.validate.callflows">>, Payload),
applications/crossbar/src/modules/cb_onboard.erl:453:    Context1 = crossbar_bindings:fold(<<"*.execute.put.accounts">>, [cb_context:set_resp_status(Context, 'error')]),
applications/crossbar/src/modules/cb_onboard.erl:470:                    catch (crossbar_bindings:fold(<<"*.execute.delete.accounts">>, [Context1, AccountId])),
applications/crossbar/src/modules/cb_onboard.erl:494:    Context1 = crossbar_bindings:fold(<<"*.execute.put.phone_numbers">>, Payload),
applications/crossbar/src/modules/cb_onboard.erl:511:    Context1 = crossbar_bindings:fold(<<"*.execute.post.braintree">>, Payload),
applications/crossbar/src/modules/cb_onboard.erl:524:    Context1 = crossbar_bindings:fold(<<"*.execute.put.", Event/binary>>, Payload),
```

I just need kazoo_bindings:map/2 except that it didn't work with wildcards. Looking at the code I thought it handles wildcards (see ^) but it didn't. So I added some code to fix that.

I understand my additions have muddied waters however I don't know how to better fix that.
Maybe remove `matches(BParts, RoutingParts)` and leave `matches(RoutingParts, BParts)`?